File: scratch/invoke_function.js
require('dotenv').config({ path: 'app/.env' });
const fetch = require('node-fetch') || globalThis.fetch;

async function run() {
  const url = process.env.SUPABASE_URL;
  const key = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_ANON_KEY;
  if (!url || !key) {
    console.error("Missing credentials");
    process.exit(1);
  }

  // Get a recent job to trigger
  const supabaseUrl = `${url}/rest/v1/jobs?select=id,tags&order=created_at.desc&limit=1`;
  const jResponse = await fetch(supabaseUrl, {
    headers: {
      'apikey': key,
      'Authorization': `Bearer ${key}`
    }
  });
  const jobs = await jResponse.json();
  if (!jobs || jobs.length === 0) {
    console.log("No jobs found");
    return;
  }
  const jobId = jobs[0].id;
  console.log(`Triggering for job ${jobId}...`);

  const funcUrl = `${url}/functions/v1/notify-job-alerts`;
  const response = await fetch(funcUrl, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${key}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ job_id: jobId })
  });

  console.log(`Status: ${response.status}`);
  const text = await response.text();
  console.log("Response:", text);
}

run().catch(console.error);

File: scratch/invoke_function.py
import os
import json
import urllib.request
from dotenv import load_dotenv

load_dotenv("app/.env")

url = os.environ.get("SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY") or os.environ.get("SUPABASE_ANON_KEY")

if not url or not key:
    print("Missing env vars")
    exit(1)

function_url = f"{url}/functions/v1/notify-job-alerts"

headers = {
    "Authorization": f"Bearer {key}",
    "Content-Type": "application/json"
}

from supabase import create_client
supabase = create_client(url, key)
res_jobs = supabase.from_('jobs').select('id, tags').order('created_at', desc=True).limit(1).execute()
if not res_jobs.data:
    print("No jobs found")
    exit(1)

job_id = res_jobs.data[0]['id']
print(f"Triggering for job {job_id}...")

payload = json.dumps({"job_id": job_id}).encode("utf-8")
req = urllib.request.Request(function_url, data=payload, headers=headers, method="POST")

try:
    with urllib.request.urlopen(req) as response:
        print(f"Status: {response.status}")
        resp_body = response.read().decode("utf-8")
        print("Response JSON:")
        print(resp_body)
except urllib.error.HTTPError as e:
    print(f"HTTPError: {e.code}")
    print(e.read().decode("utf-8"))
except Exception as e:
    print(f"Error: {e}")
